gdc_ng_models/models/cohort.py

```diff
@@ -0,0 +1,222 @@
+"""
+
+import uuid
+
+from sqlalchemy import orm, ARRAY, BigInteger, Boolean, Column, ForeignKey, Text
+from sqlalchemy.ext import declarative
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+from gdc_ng_models.models import audit
+
+Base = declarative.declarative_base()
+
+
+class AnonymousContext(Base, audit.AuditColumnsMixin):
+    """An anonymous context to drive authorization for cohort manipulation.
+
+    Provides authorization capabilities in lieu of a login (something the GDC
+    data portal does not and cannot implement). When a cohort is created, it
+    is associated with a context ID that establishes identity similar to a
+    bearer token. When a request is made to manipulate a cohort, the context ID
+    must be provided as part of that request to authorize a change.
+
+    Attributes:
+        id: A UUID identifier for the context.
+        name: A human-readable name for the context.
+        created_datetime: The date and time when the record is created.
+        updated_datetime: The date and time when the record is updated.
+    """
+
+    __tablename__ = "anonymous_context"
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    name = Column(Text, nullable=False)
+
+    # establishes a one-to-many relationship with Cohort
+    cohorts = orm.relationship("Cohort")
+
+    def __repr__(self):
+        return (
+            "<AnonymousContext("
+            "id={id}, "
+            "name='{name}', "
+            "created_datetime={created_datetime}, "
+            "updated_datetime={updated_datetime})>".format(
+                id=self.id,
+                name=self.name,
+                created_datetime=self.created_datetime.isoformat() if self.created_datetime else None,
+                updated_datetime=self.updated_datetime.isoformat() if self.updated_datetime else None,
+            )
+        )
+
+    def to_json(self):
+        return {
+            "id": str(self.id),
+            "name": self.name,
+            "created_datetime": self.created_datetime.isoformat() if self.created_datetime else None,
+            "updated_datetime": self.updated_datetime.isoformat() if self.updated_datetime else None,
+        }
+
+
+class Cohort(Base, audit.AuditColumnsMixin):
+    """A base definition for a cohort entity.
+
+    Attributes:
+        id: A UUID identifier for the cohort.
+        context_id: The UUID of the associated context.
+        name: A user defined name for the cohort.
+        created_datetime: The date and time when the record is created.
+        updated_datetime: The date and time when the record is updated.
+    """
+
+    __tablename__ = "cohort"
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    name = Column(Text, nullable=False)
+    context_id = Column(UUID(as_uuid=True), ForeignKey("anonymous_context.id"), nullable=False)
+
+    # establishes a one-to-many relationship with CohortFilter
+    filters = orm.relationship("CohortFilter")
```

Comment:
needs `back_populates` and potentially loading strategy

Reply:
Thank you for the suggestion! This is done.